Content.Server/GameObjects/Components/Weapon/Melee/MeleeWeaponComponent.cs

```diff
@@ -17,79 +17,78 @@
+        public float CooldownTime { get; private set; } = 0.5f;
 
         [ViewVariables(VVAccess.ReadWrite)]
-        public string Arc
-        {
-            get => _arc;
-            set => _arc = value;
-        }
+        public string ClickArc { get; set; }
 
         [ViewVariables(VVAccess.ReadWrite)]
-        public float ArcWidth
-        {
-            get => _arcWidth;
-            set => _arcWidth = value;
-        }
+        public string Arc { get; set; }
 
         [ViewVariables(VVAccess.ReadWrite)]
-        public float Range
-        {
-            get => _range;
-            set => _range = value;
-        }
+        public float ArcWidth { get; set; }
 
         [ViewVariables(VVAccess.ReadWrite)]
-        public int Damage
-        {
-            get => _damage;
-            set => _damage = value;
-        }
+        public float Range { get; set; }
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public int Damage { get; set; }
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public DamageType DamageType { get; set; }
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public bool ClickAttackEffect { get; set; }
 
         public override void ExposeData(ObjectSerializer serializer)
         {
             base.ExposeData(serializer);
 
-            serializer.DataField(ref _damage, "damage", 5);
-            serializer.DataField(ref _range, "range", 1);
-            serializer.DataField(ref _arcWidth, "arcwidth", 90);
-            serializer.DataField(ref _arc, "arc", "default");
-            serializer.DataField(ref _hitSound, "hitSound", "/Audio/Weapons/genhit1.ogg");
-            serializer.DataField(ref _cooldownTime, "cooldownTime", 1f);
+            serializer.DataField(this, x => x.Damage, "damage", 5);
+            serializer.DataField(this, x => x.Range, "range", 1);
+            serializer.DataField(this, x => x.ArcWidth, "arcwidth", 90);
+            serializer.DataField(this, x => x.Arc, "arc", "default");
+            serializer.DataField(this, x => x.ClickArc, "clickArc", "punch");
+            serializer.DataField(this, x => x._hitSound, "hitSound", "/Audio/Weapons/genhit1.ogg");
+            serializer.DataField(this, x => x._missSound, "hitSound", "/Audio/Weapons/punchmiss.ogg");
+            serializer.DataField(this, x => x.ArcCooldownTime, "arcCooldownTime", 1f);
+            serializer.DataField(this, x => x.CooldownTime, "cooldownTime", 1f);
+            serializer.DataField(this, x => x.DamageType, "damageType", DamageType.Blunt);
+            serializer.DataField(this, x => x.ClickAttackEffect, "clickAttackEffect", true);
         }
 
         protected virtual bool OnHitEntities(IReadOnlyList<IEntity> entities, AttackEventArgs eventArgs)
         {
             return true;
         }
 
-        void IAttack.Attack(AttackEventArgs eventArgs)
+        bool IAttack.WideAttack(AttackEventArgs eventArgs)
         {
+            if (!eventArgs.WideAttack) return true;
```

Comment:
This should return false to let other IAttack interfaces to run... Even though the return type isn't used anywhere yet. Just for consistency with the rest of the interaction interfaces.